### ISSUE TITLE
Auto-correction breaks code by `Lint/ParenthesesAsGroupedExpression`

### ISSUE THREAD

**Issue by universato** [rubocop/rubocop#11192]
# Steps to reproduce the problem

I ran `rubocop -a --only Lint/ParenthesesAsGroupedExpression` on the following codes.

```ruby
a = [1, 2]
a.concat (1..2).map { |i| i * 10 }
p a
```

the fix breaks the code.

## Expected behavior

```ruby
a.concat (1..1).map { |i| i * 10 }   # Either leave it as it is,
a.concat((1..2).map { |i| i * 10 })  # or add extra parenthesis
```

## Actual behavior

```ruby
a.concat(1..2).map { |i| i * 10 } # It removes the space
```

```
$ rubocop --debug -a concat.rb
For ~/til: configuration from ~/til/.rubocop.yml
Default configuration from ~/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rubocop-1.39.0/config/default.yml
AllCops/Exclude configuration from ~/.rubocop.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning ~/til/concat.rb
Loading cache from ~/.cache/rubocop_cache/a2565da9ab78907e6417c25d41adf90bb6c76cd3/6d7a3b621ca1730e04accd938619e4bdab66cfb1/711fe9505675c0238242e0fdfccd685e0db9a88d
W

Offenses:

concat.rb:2:9: W: [Corrected] Lint/ParenthesesAsGroupedExpression: (1..2).map { |i| i * 10 } interpreted as grouped expression.
a.concat (1..2).map { |i| i * 10 }
        ^

1 file inspected, 1 offense detected, 1 offense corrected
Finished in 0.22206000000005588 seconds
```
## RuboCop version

```console
$ rubocop -V
1.39.0 (using Parser 3.1.2.1, rubocop-ast 1.23.0, running on ruby 3.1.0) [x86_64-darwin20]
```
